connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java

```diff
@@ -117,8 +182,10 @@ protected float getDrag(GeyserSession session) {
      */
     protected boolean isInWater(GeyserSession session) {
         if (session.getConnector().getConfig().isCacheChunks()) {
-            int block = session.getConnector().getWorldManager().getBlockAt(session, position.toInt());
-            return block == BlockTranslator.BEDROCK_WATER_ID;
+            if (0 <= position.getFloorY() && position.getFloorY() <= 255) {
```

Comment:
Just as a general note, it might be a better idea to move this as a precheck for getBlockAt. You can get a similar error if you place a block at Y256 because of the resulting block update packet. No need to change this code for the time being.